torchvision/datasets/kitti.py

```diff
@@ -0,0 +1,184 @@
+        "data_object_image_2.zip",
+        "data_object_label_2.zip",
+    ]
+    image_dir_name = "image_2"
+    labels_dir_name = "label_2"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = None,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        transforms: Optional[Callable] = None,
+        download: bool = False,
+    ):
+        super().__init__(
+            root,
+            transform=transform,
+            target_transform=target_transform,
+            transforms=transforms,
+        )
+        self.TargetTuple = namedtuple(
+            "TargetTuple",
+            [
+                "type",
+                "truncated",
+                "occluded",
+                "alpha",
+                "bbox",
+                "dimensions",
+                "location",
+                "rotation_y",
+            ],
+        )
+        self.images = []
+        self.targets = []
+        self.root = root
+        self.split = split
+        self.transform = transform
+        self.target_transform = target_transform
+        self.transforms = transforms
+
+        if download:
+            self.download()
+        if not self._check_exists():
+            raise RuntimeError(
+                "Dataset not found. You may use download=True to download it."
+            )
+
+        location = "testing" if self.split == "test" else "training"
+        image_dir = os.path.join(self.raw_folder, location, self.image_dir_name)
+        if location == "training":
+            labels_dir = os.path.join(self.raw_folder, location, self.labels_dir_name)
+        for img_file in os.listdir(image_dir):
+            self.images.append(os.path.join(image_dir, img_file))
+            if location == "training":
+                self.targets.append(
+                    os.path.join(labels_dir, f"{img_file.split('.')[0]}.txt")
+                )
+
+    def __getitem__(self, index: int) -> Tuple[Any, Any]:
+        """Get item at a given index.
+        Args:
+            index (int): Index
+        Returns:
+            tuple: (image, target), where
+            target is a namedtuple with the following fields:
+                type: Int64Tensor[N]
+                truncated: FloatTensor[N]
+                occluded: Int64Tensor[N]
+                alpha: FloatTensor[N]
+                bbox: FloatTensor[N, 4]
+                dimensions: FloatTensor[N, 3]
+                locations: FloatTensor[N, 3]
+                rotation_y: FloatTensor[N]
+                score: FloatTensor[N]
+        """
+        image = Image.open(self.images[index])
+        target = None if self.split == "test" else self._parse_target(index)
+        if self.transforms:
+            image, target = self.transforms(image, target)
+        if self.transform:
+            image = self.transform(image)
+        if self.target_transform:
+            target = self.target_transform(target)
```

Comment:
Instead of having both `transforms` as well as `transform` and `target_transform`, I propose to instead only have a single `transforms` call.

In the current implementation, we would be calling the transforms twice, as `VisionDataset` provides an automatic replacement for `transform` and `transforms` in https://github.com/pytorch/vision/blob/f9af70a98244b30f645fb8a033b6e4ce59ca020f/torchvision/datasets/vision.py#L27-L29

This is also what other datasets similar to this one (like COCO and VOC) do, see https://github.com/pytorch/vision/blob/f9af70a98244b30f645fb8a033b6e4ce59ca020f/torchvision/datasets/coco.py#L48-L49 for example